### PR TITLE
prov/rxd: Removed redundant memset() calls

### PR DESCRIPTION
**Pull request by iherna2**
Signed-off-by: Ignacio Hernandez <ignacio.hernandez@intel.com>

### CODE REVIEW

File: prov/rxd/src/rxd_ep.c [hunk 1]
```diff
@@ -453,7 +455,7 @@ struct rxd_x_entry *rxd_tx_entry_init(struct rxd_ep *ep, const struct iovec *iov
 	memcpy(&tx_entry->iov[0], iov,
 	       sizeof(*iov) * iov_count);
 
-	memset(&tx_entry->cq_entry, 0, sizeof(tx_entry->cq_entry));
+	tx_entry->cq_entry.data = 0;
```
Comment (aingerson) [inline prov/rxd/src/rxd_ep.c]: Trivial, but I think we can save an instruction with
`tx_entry->cq_entry.data = (flags & FI_REMOTE_CQ_DATA) ? data : 0`

File: prov/rxd/src/rxd_ep.c [hunk 2]
```diff
@@ -550,7 +552,6 @@ static ssize_t rxd_ep_post_rts(struct rxd_ep *rxd_ep, struct rxd_x_entry *tx_ent
 	rxd_init_ctrl_pkt(rxd_ep, tx_entry, pkt_entry, RXD_RTS);
 	rts_pkt = rxd_get_ctrl_pkt(pkt_entry);
 	addrlen = RXD_NAME_LENGTH;
-	memset(rts_pkt->ctrl_hdr.source, 0, RXD_NAME_LENGTH);
```
Comment (aingerson) [inline prov/rxd/src/rxd_ep.c]: I'm assuming you tested these changes with the pingpong and bandwidth benchmarks? I fixed an old bug in the bandwidth test with this memset but I think it was actually related to a separate bug fix that I submitted later.

Comment (iherna2) [reply prov/rxd/src/rxd_ep.c]: Interesting...., I wonder if the bug was elsewhere as getname should overwrite the string right after.

I didn't test bandwidth I'll do so and report back

File: prov/rxd/src/rxd_ep.c (hunk 1 above)
Comment (shefty) [reply prov/rxd/src/rxd_ep.c]: Is this setting up the CQ entry for a transmit request?

Comment (iherna2) [reply prov/rxd/src/rxd_ep.c]: Yes, I'll post an update

Comment (iherna2) [reply prov/rxd/src/rxd_ep.c]: Yes, that sounds like a good idea

File: prov/rxd/src/rxd_ep.c (hunk 2 above)
Comment (iherna2) [reply prov/rxd/src/rxd_ep.c]: Interesting, let me test bandwidth, the string should be overwritten right after by fi_getname() call. I'm wondering if there is an error elsewhere.

File: prov/rxd/src/rxd_ep.c (hunk 1 above)
Comment (shefty) [reply prov/rxd/src/rxd_ep.c]: The cq data for a transmit shouldn't matter.

**Comment by shefty** [comment]
Is there any update on this PR?

**Comment by aingerson** [comment]
I included these changes in the recent per peer management updates. Closing this PR

**Pull request closed without merging**